Assets/MixedRealityToolkit.ThirdParty/MRTK-Quest/Scripts/Input/Controllers/OculusQuestHand.cs

```diff
@@ -84,6 +84,13 @@ public OculusQuestHand(TrackingState trackingState, Handedness controllerHandedn
             this.handMaterial = handMaterial;
             handRenderer = ovrHand.GetComponent<Renderer>();
 
+            UpdateHandMaterial(handMaterial);
+
+        }
+
+        public void UpdateHandMaterial(Material handMaterial)
+        {
+
             if (!MRTKOculusConfig.Instance.UseCustomHandMaterial) return;
 
```

Comment:
Please update this.handMaterial when updating the material

Reply:
Currently, `this.handMaterial` is a read-only private variable. I'm unsure if there is any butterfly effect of changing it. 

This function `UpdateHandMaterial()` updates the material to the current instance of the hand by directly tapping into the renderer. When the tracking is lost and regained, a new instance of quest hand will be created and with that `this.handMaterial` will store the updated material. 

I can see a drawback, where if any code which will try to read the `handMaterial` will receive old material for that particular instance.

Reply:
You definitely need to update this.handMaterial, or the pinch affordance won’t work until the new hand is made.

It’s read only because it wasn’t replaced at runtime before, but it’s not that big a deal to change it.

Reply:
Thank you for your guidance. Made necessary changes.